email.go

```diff
@@ -433,6 +433,34 @@ func (e *Email) AttachFile(filename string) (a Attachment, err error) {
 	return e.Attach(f, basename, ct)
 }
 
+// formatAddress formats the address as a valid RFC 5322 address.
+// If the address's name contains non-ASCII characters
+// the name will be rendered according to RFC 2047.
+func (e *Email) formatAddress(addr string) (string, error) {
+	em, err := mail.ParseAddress(addr)
+	if err != nil {
+		return addr, err
+	}
+
+	return em.String(), nil
+}
+
+// formatAddresses formats the list of address as a valid RFC 5322 addresses.
+// If the address's name contains non-ASCII characters
+// the name will be rendered according to RFC 2047.
+func (e *Email) formatAddresses(addrs []string) ([]string, error) {
+	fAddrs := make([]string, len(addrs)-1)
```

Comment:
The append() below will insert new items on top of the allocated length. Should instead assign the capacity here like: `make([]string, 0, len(addrs)-1)`.

Reply:
Makes sense. Updated the PR.

Reply:
Sorry, missed this. It should be len(addrs) and not len(addrs)-1, right?

Reply:
Ah, elementary mistake. Fixed it.